### PR TITLE
Use Pydantic's BaseSettings for project configuration

### LINKED ISSUE #124: Use pydantic's BaseSettings for configuration instead of startlette's config
FastAPI recommends using BaseSettings for confiugration. It would be nice for this example to follow that recommendation. See https://fastapi.tiangolo.com/advanced/settings/

### PR DESCRIPTION
**Pull request by samuelfirst**
Fixes #124

### CODE REVIEW

**Review by nsidnev — CHANGES_REQUESTED** [review]
@samuelfirst thanks so much for this PR! There are a few changes that need to be fixed before this can be merged. Could you take a look at the comments?

File: app/core/settings/app.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+class AppSettings(BaseAppSettings):
+    debug: bool = False
+    docs_url: str = "/docs"
+    openapi_prefix: str = ""
+    openapi_url: str = "/openapi.json"
+    redoc_url: str = "/redoc"
+    title: str = "FastAPI example application"
+    version: str = "0.0.0"
+
+    database_url: PostgresDsn
```
Comment (nsidnev) [inline app/core/settings/app.py:17]: I think this also needs to be changed in the GitHub Actions workflow files so that they complete successfully.

**Review by nsidnev — CHANGES_REQUESTED** [review]

File: README.rst [hunk 2]
```diff
@@ -46,7 +46,8 @@ Then run the following commands to bootstrap your environment with ``poetry``: :
 Then create ``.env`` file (or rename and modify ``.env.example``) in project root and set environment variables for application: ::
 
     touch .env
-    echo DB_CONNECTION=postgresql://$POSTGRES_USER:$POSTGRES_PASSWORD@$POSTGRES_HOST:$POSTGRES_PORT/$POSTGRES_DB >> .env
+    echo APP_ENC=dev
```
[the remark this answers is not in the document]
Comment (samuelfirst) [reply README.rst:49]: Missed this typo(

File: app/api/dependencies/authentication.py [hunk 3]
```diff
@@ -65,19 +66,21 @@ def _get_authorization_header_optional(
...
 
 
 async def _get_current_user(
     users_repo: UsersRepository = Depends(get_repository(UsersRepository)),
     token: str = Depends(_get_authorization_header_retriever()),
+    settings: AppSettings = Depends(get_app_settings),
 ) -> User:
     try:
-        username = jwt.get_username_from_token(token, str(SECRET_KEY))
+        username = jwt.get_username_from_token(token, str(settings.secret_key))
```
[the remark this answers is not in the document]
Comment (samuelfirst) [reply app/api/dependencies/authentication.py:83]: You are right!)

File: app/core/config.py [hunk 4]
```diff
@@ -1,44 +1,20 @@
...
+from app.core.settings.test import TestAppSettings
+
+environments: Dict[str, Type[AppSettings]] = {
+    AppEnvTypes.dev: DevAppSettings,
+    AppEnvTypes.prod: ProdAppSettings,
+    AppEnvTypes.test: TestAppSettings,
+}
+
+
+def get_app_settings() -> AppSettings:
```
[the remark this answers is not in the document]
Comment (samuelfirst) [reply app/core/config.py:17]: `@lru_cache` was here but for some reason I removed it and forgot to return. I'll add `@cache`

File: app/core/settings/production.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+from app.core.settings.app import AppSettings
+
+
+class ProdAppSettings(AppSettings):
```
[the remark this answers is not in the document]
Comment (samuelfirst) [reply app/core/settings/production.py:4]: Yep, I can)

File: app/core/settings/test.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+from app.core.settings.app import AppSettings
+
+
+class TestAppSettings(AppSettings):
+    debug: bool = True
+
+    title: str = "Test FastAPI example application"
+
+    secret_key: str = "test_secret"
```
[the remark this answers is not in the document]
Comment (samuelfirst) [reply app/core/settings/test.py:13]: I'll change it to `secret_key: SecretStr = SecretStr("test_secret")`

File: app/core/settings/app.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    secret_key: SecretStr
+
+    api_prefix: str = "/api"
+
+    jwt_token_prefix: str = "Token"
+
+    allowed_hosts: Union[str, List[str]] = Field(..., env="ALLOWED_HOSTS")
+
+    @validator("allowed_host", pre=True, check_fields=False)
```
[the remark this answers is not in the document]
Comment (samuelfirst) [reply app/core/settings/app.py:29]: Just one more typo.

**Review by nsidnev — APPROVED** [review]

**Comment by nsidnev** [comment]
Thanks!

**Pull request merged**